$NetBSD: patch-ac,v 1.1 2008/01/29 04:18:44 bjs Exp $

XftFontOpenInfo: Use of uninitialised value of size 8 (bug 11200)
(from freedesktop.org GIT)

--- src/xftfreetype.c.orig	2006-12-07 05:17:19.000000000 -0500
+++ src/xftfreetype.c
@@ -392,6 +392,13 @@ XftFontInfoFill (Display *dpy, _Xconst F
 	return FcFalse;
 
     /*
+     * Initialize the whole XftFontInfo so that padding doesn't interfere with
+     * hash or XftFontInfoEqual().
+     */
+ 
+    memset (fi, '\0', sizeof(*fi));
+
+    /*
      * Find the associated file
      */
     switch (FcPatternGetString (pattern, FC_FILE, 0, &filename)) {
@@ -419,8 +426,6 @@ XftFontInfoFill (Display *dpy, _Xconst F
     else if (FcPatternGetFTFace (pattern, FC_FT_FACE, 0, &face) == FcResultMatch
 	     && face)
 	fi->file = _XftGetFaceFile (face);
-    else
-	fi->file = 0;
     if (!fi->file)
         goto bail0;
 
